src/module.c

```diff
@@ -11224,7 +11230,8 @@ int RM_ScanKey(RedisModuleKey *key, RedisModuleScanCursor *cursor, RedisModuleSc
                 p = lpNext(lp, p);
 
                 /* Skip expired fields */
-                if (hashTypeIsExpired(o, vllExpire))
+                if ((hashTypeIsExpired(o, vllExpire)) &&
+                    (!(key->mode & REDISMODULE_OPEN_KEY_ACCESS_EXPIRED)))
```

Comment:
@moticless Consider replacing the order of the expressions

Reply:
The same claim form above goes here

Reply:
Looking at it again, actually because we have here && and it check both conditions anyway, it is worth to exchange.